tests/unit/s2n_server_key_share_extension_test.c

```diff
@@ -885,15 +885,25 @@ int main(int argc, char **argv)
                     S2N_STUFFER_READ_EXPECT_EQUAL(&stuffer, kem_group->iana_id, uint16);
                     S2N_STUFFER_READ_EXPECT_EQUAL(&stuffer, expected_hybrid_share_size, uint16);
 
-                    if (len_prefixed) {
-                        S2N_STUFFER_READ_EXPECT_EQUAL(&stuffer, kem_group->curve->share_size, uint16);
-                    }
-                    EXPECT_SUCCESS(s2n_stuffer_skip_read(&stuffer, kem_group->curve->share_size));
-
-                    if (len_prefixed) {
-                        S2N_STUFFER_READ_EXPECT_EQUAL(&stuffer, kem_group->kem->ciphertext_length, uint16);
+                    if (kem_group->send_kem_first) {
+                        if (len_prefixed) {
+                            S2N_STUFFER_READ_EXPECT_EQUAL(&stuffer, kem_group->kem->ciphertext_length, uint16);
+                        }
+                        EXPECT_SUCCESS(s2n_stuffer_skip_read(&stuffer, kem_group->kem->ciphertext_length));
+                        if (len_prefixed) {
+                            S2N_STUFFER_READ_EXPECT_EQUAL(&stuffer, kem_group->curve->share_size, uint16);
+                        }
+                        S2N_STUFFER_LENGTH_WRITTEN_EXPECT_EQUAL(&stuffer, kem_group->curve->share_size);
+                    } else {
```

Comment:
This test was already pretty complex, and I think this tips it over into problematically complex. Can you simplify this test so that it's more obviously correct? Like, maybe it's time to test each of these cases separately rather than in one big branching test.

Reply:
Simplified the logic here.